etc/esp-idf/CMakeLists.txt

```diff
@@ -0,0 +1,159 @@
+    $ENV{OPENMRNPATH}/src/openlcb/Velocity.cxx
+    $ENV{OPENMRNPATH}/src/openlcb/WriteHelper.cxx
+
+    
+    $ENV{OPENMRNPATH}/src/os/FakeClock.cxx
+    $ENV{OPENMRNPATH}/src/os/logging_malloc.cxx
+    $ENV{OPENMRNPATH}/src/os/MDNS.cxx
+    $ENV{OPENMRNPATH}/src/os/os.c
+    $ENV{OPENMRNPATH}/src/os/OSImpl.cxx
+    $ENV{OPENMRNPATH}/src/os/OSSelectWakeup.cxx
+    $ENV{OPENMRNPATH}/src/os/stack_malloc.c
+    $ENV{OPENMRNPATH}/src/os/TempFile.cxx
+    $ENV{OPENMRNPATH}/src/os/watchdog.c
+
+    $ENV{OPENMRNPATH}/src/utils/Base64.cxx
+    $ENV{OPENMRNPATH}/src/utils/Blinker.cxx
+    $ENV{OPENMRNPATH}/src/utils/Buffer.cxx
+    $ENV{OPENMRNPATH}/src/utils/CanIf.cxx
+    $ENV{OPENMRNPATH}/src/utils/ClientConnection.cxx
+    $ENV{OPENMRNPATH}/src/utils/ConfigUpdateListener.cxx
+    $ENV{OPENMRNPATH}/src/utils/constants.cxx
+    $ENV{OPENMRNPATH}/src/utils/Crc.cxx
+    $ENV{OPENMRNPATH}/src/utils/DirectHub.cxx
+    $ENV{OPENMRNPATH}/src/utils/DirectHubGc.cxx
+    $ENV{OPENMRNPATH}/src/utils/DirectHubLegacy.cxx
+    $ENV{OPENMRNPATH}/src/utils/errno_exit.c
+    $ENV{OPENMRNPATH}/src/utils/FdUtils.cxx
+    $ENV{OPENMRNPATH}/src/utils/FileUtils.cxx
+    $ENV{OPENMRNPATH}/src/utils/format_utils.cxx
+    $ENV{OPENMRNPATH}/src/utils/ForwardAllocator.cxx
+    $ENV{OPENMRNPATH}/src/utils/GcStreamParser.cxx
+    $ENV{OPENMRNPATH}/src/utils/GcTcpHub.cxx
+    $ENV{OPENMRNPATH}/src/utils/gc_format.cxx
+    $ENV{OPENMRNPATH}/src/utils/GridConnect.cxx
+    $ENV{OPENMRNPATH}/src/utils/GridConnectHub.cxx
+    $ENV{OPENMRNPATH}/src/utils/HubDevice.cxx
+    $ENV{OPENMRNPATH}/src/utils/HubDeviceSelect.cxx
+    $ENV{OPENMRNPATH}/src/utils/ieeehalfprecision.c
+    $ENV{OPENMRNPATH}/src/utils/JSHubPort.cxx
+    $ENV{OPENMRNPATH}/src/utils/logging.cxx
+    $ENV{OPENMRNPATH}/src/utils/Queue.cxx
+    $ENV{OPENMRNPATH}/src/utils/ReflashBootloader.cxx
+    $ENV{OPENMRNPATH}/src/utils/ServiceLocator.cxx
+    $ENV{OPENMRNPATH}/src/utils/SocketCan.cxx
+    $ENV{OPENMRNPATH}/src/utils/SocketClient.cxx
+    $ENV{OPENMRNPATH}/src/utils/socket_listener.cxx
+    $ENV{OPENMRNPATH}/src/utils/Stats.cxx
+    $ENV{OPENMRNPATH}/src/utils/StringPrintf.cxx
+)
+
+set(INCLUDE_DIRS
+    $ENV{OPENMRNPATH}/src
+    $ENV{OPENMRNPATH}/include/esp-idf
+    $ENV{OPENMRNPATH}/include
+)
+
+if(CONFIG_ENABLE_ARDUINO_DEPENDS)
+    list(APPEND SOURCE_DIRS src)
+endif()
+
+set(IDF_DEPS
+    app_update
+    bootloader_support
+    driver
+    esp_adc
+    esp_app_format
+    esp_system
+    esp_wifi
+    espcoredump
+    hal
+    heap
+    vfs
+    mdns
+    bt)
+
+idf_component_register(SRCS "${SRCS}"
+                       INCLUDE_DIRS "${INCLUDE_DIRS}"
+                       REQUIRES "${IDF_DEPS}")
+
+###############################################################################
+# Suppress compilation warnings in OpenMRN
+###############################################################################
+
+target_compile_options(${COMPONENT_LIB} PUBLIC $<$<COMPILE_LANGUAGE:CXX>:-Wno-volatile>)
```

Comment:
Does this need to be PUBLIC?

Reply:
I opted for `PUBLIC` primarily to suppress a warning from Executor.hxx/cxx showing up if you include the hxx in your project. We might be able to get away with `INTERFACE` though, I'll check and confirm.

Reply:
As noted elsewhere, we can reduce this to `PRIVATE`

Reply:
This is required because of a warning generated by the usage of Executor::sequence_. We should revisit this at some point.